processor/probabilisticsamplerprocessor/config.go

```diff
@@ -15,26 +15,50 @@
 package probabilisticsamplerprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor"
 
 import (
+	"fmt"
+
 	"go.opentelemetry.io/collector/config"
 )
 
-// Config has the configuration guiding the trace sampler processor.
+type AttributeSource string
+
+const (
+	traceIDAttributeSource = AttributeSource("traceID")
+	recordAttributeSource  = AttributeSource("record")
+
+	defaultAttributeSource = traceIDAttributeSource
+)
+
+// Config has the configuration guiding the sampler processor.
 type Config struct {
 	config.ProcessorSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 
-	// SamplingPercentage is the percentage rate at which traces are going to be sampled. Defaults to zero, i.e.: no sample.
-	// Values greater or equal 100 are treated as "sample all traces".
+	// SamplingPercentage is the percentage rate at which traces or logs are going to be sampled. Defaults to zero, i.e.: no sample.
+	// Values greater or equal 100 are treated as "sample all traces/logs".
 	SamplingPercentage float32 `mapstructure:"sampling_percentage"`
 
 	// HashSeed allows one to configure the hashing seed. This is important in scenarios where multiple layers of collectors
 	// have different sampling rates: if they use the same seed all passing one layer may pass the other even if they have
 	// different sampling rates, configuring different seeds avoids that.
 	HashSeed uint32 `mapstructure:"hash_seed"`
+
+	// AttributeSource (logs only) defines where to look for the attribute in from_attribute. The allowed values are
+	// `traceID` or `record`. Default is `traceID`.
+	AttributeSource `mapstructure:"attribute_source"`
+	// FromAttribute (logs only) The optional name of a log record attribute used for sampling purposes, such as a
```

Comment:
If you are changing this PR for any other reason, please add an empty line between lines 47 and 48, and between 50-51.